internal/helm/repository/endpoint.go

```
@@ -0,0 +1,97 @@
+package repository
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/asaskevich/govalidator.v8"
+
+	"github.com/supergiant/supergiant/pkg/model/helm"
+)
+
+// Controller is a http controller for a helm repositories.
+type Controller struct {
+	svc *Service
+}
+
+// New constructs a Controller for helm repositories.
+func New(svc *Service) (*Controller, error) {
```

Please change to NewService and remove error in return

---

I'll rename it. BTW, it would be better to keep a service and an endpoint separately.

About an error, I've supposed it is a stable pattern for a constructor: __New(opts) (Obj, error)__, because now you might not have an error, but WHEN you will add some opts and you need to return an error, you should updated/modify code where it used.
The reason to add an error now, that it isn't affect the complexity.